src/bin/genemichaels.rs

```diff
@@ -196,13 +232,103 @@ fn main() {
-        src.lines().take(5).any(|l| l.contains("`nogenemichaels`"))
-    }
+                // bold green
+                eprint!("\x1B[1;32m");
+                eprint!("  Formatting ");
 
-    if args.files.is_empty() {
-        match es!({
+                // reset
+                eprint!("\x1B[0;22m");
+                eprintln!("workspace...");
+                let mut cargo_toml = None;
+
+                // find workspace cargo toml
+                for parent in 0.. {
+                    let dir = if parent == 0 {
+                        ".".into()
+                    } else {
+                        "../".repeat(parent)
+                    };
+                    match std::fs::read_dir(dir)?.filter_map(|f| f.ok()).find(|f| f.file_name() == "Cargo.toml") {
+                        Some(path) if
+                            cargo_manifest::Manifest::from_path(path.path())
+                                .ok()
+                                .and_then(|m| m.workspace)
+                                .is_some() => {
+                            cargo_toml = Some(path);
+                            break;
+                        },
+                        _ => {
+                            continue;
+                        },
+                    }
+                }
+                let path = cargo_toml.ok_or_else(|| anyhow::anyhow!("No workspace found!"))?.path().canonicalize()?;
+                let workspace_folder = &path.parent().unwrap();
+                let manifest = cargo_manifest::Manifest::from_path(&path)?;
+                let mut dirs = manifest.workspace.ok_or_else(|| anyhow::anyhow!("No workspace found!"))?.members;
+                if fs::read_dir("./src").is_ok() {
+                    // checks current dir too if src folder exists
+                    dirs.push(String::new());
+                };
+                let mut threads = Vec::new();
+                for dir in dirs {
+                    let mut path = workspace_folder.to_path_buf();
+                    path.push(dir);
+                    for f in walkdir::WalkDir::new(&path) {
+                        match f {
+                            Ok(file) => {
+                                let file_path = file.path().to_path_buf();
+                                if file_path.extension() == Some(OsStr::new("rs")) &&
+                                    !file_path.components().any(|c| c.as_os_str() == OsStr::new("target")) {
+                                    {
+                                        threads.push(
+                                            thread::spawn(
+                                                move || process_workspace_file(config, file.path().to_path_buf()),
+                                            ),
+                                        );
+                                    }
+                                }
+                            },
+                            Err(_) => {
```

Comment:
I have no idea what this could be, but I think it would be good to warn here just so people aren't blind to potentially confusing issues.